cpp/include/cudf/detail/sizes_to_offsets_iterator.cuh

```diff
@@ -269,5 +273,55 @@ auto sizes_to_offsets(SizesIterator begin,
   return last_element.value(stream);
 }
 
+/**
+ * @brief Create an offsets column to be a child of a compound column
+ *
+ * This function sets the offsets values by executing scan over the sizes in the provided
+ * Iterator.
+ *
+ * The return also includes the total number of elements -- the last element value from the
+ * scan.
+ *
+ * @throw cudf::logic_error if the total size of the scan (last element) greater than maximum value
+ * of `size_type`
+ *
+ * @tparam Iterator Used as input to scan to set the offset values
```

Comment:
```suggestion
 * @tparam InputIterator Used as input to scan to set the offset values
```

Reply:
Wonder why this did not fail the doxygen check.